### ISSUE TITLE
Panic when clicking on certain widgets after state change

### ISSUE THREAD

**Issue by MAlba124** [slint-ui/slint#10269]
### Bug Description

When clicking either the `LineEdit` or `Slider` after clicking on the button to change state in the code below, the program crashes with this panic:

```console
thread 'main' (540470) panicked at [...]/accesskit_consumer-0.31.0/src/tree.rs:34:13:
Focused ID #12884901890 is not in the node list
```

main.rs:

```rust
slint::include_modules!();

fn main() {
    let ui = MainWindow::new().unwrap();

    let ui_weak = ui.as_weak();

    ui.global::<Glob>().on_rust_function(move || {
        // Not bugged:
        // ui_weak.upgrade().unwrap().global::<Glob>().invoke_change_state();

        // Bugged:
        ui_weak.upgrade_in_event_loop(|ui| {
            ui.global::<Glob>().invoke_change_state();
        }).unwrap();
    });

    ui.run().unwrap();
}
```

### Reproducible Code (if applicable)

```slint
import { Button, Slider, LineEdit, } from "std-widgets.slint";

export enum AppState {
    Foo,
    Bar,
}

export global Glob {
    in-out property <AppState> app-state: Foo;
    callback rust-function();
    public function change-state() {
        app-state = AppState.Bar;
    }
}

component Component {
    HorizontalLayout {
        if Glob.app-state == AppState.Foo: Rectangle {
            Button {
                text: "Click me";

                clicked => Glob.rust-function();
            }
        }

        if Glob.app-state == AppState.Bar: Rectangle {
            HorizontalLayout {
                i-line-edit-url := LineEdit {
                    placeholder-text: "Click me";
                }

                Slider {
                    width: 20px;
                }
            }
        }
    }
}

export component MainWindow inherits Window {
    Component { }
}
```

### Environment Details

- Slint Version: 1.15
- Platform/OS: linux wayland
- Programming Language: rust
- Backend/Renderer: winit-femtovg and winit-software

**Comment by tronical** [comment]
I can reproduce this also on macOS (note: VoiceOver or the accessibility inspector need to be enabled).

**Comment by tronical** [comment]
As far as I can see, the issue is this:

1. Clicking the button triggers a (delayed) state change. The Button is to be destroyed and the LineEdit/Slider to be created.
2. The tracker of some existing access kit node is dirty (maybe the Button state somehow? To be investigated...), so eventually `rebuild_tree_of_dirty_nodes()` is called from a single-shot timer. This mutably borrows the access kit adapter and begins a rebuild of only the changed access kit nodes.
3. During the rebuild, `ItemRc::is_visible()` is called, which ends up traversing the item tree up and triggering an update of the repeater in the example.
4. The LineEdit/Slider is created, `register_item_tree()` is called in the `winitwindowadapter.rs` - we need to create access kit nodes for these.
5. However, that function returns as the attempt to mutably borrow the access kit adapter fails. (see (2))
6. Later, focus is transferred to the line edit, and we report a node id to access kit that it doesn't know about and (rightly so) aborts.

**Comment by tronical** [comment]
This is tricky to fix, I think we need to re-work the interior mutability model of the access kit adapter: Exclusive access to the adapter as a whole from the window adapter isn't working. I'm thinking that we may have to introduce fine-grained interior mutability in the adapter and then handle `rebuild_tree_of_dirty_nodes()` differently:

Normally it would send a `TreeUpdate { nodes: changed_nodes, tree: None, focus_node }`, but if during the update of the dirty nodes we receive a `register_item_tree()` we may have to thrown the partial updates away and do a full update:

```rust
self.global_property_tracker.as_ref().evaluate_as_dependency_root(|| {
   ...
});

if self.tree_rebuild_requested.get() { // <-- new condition
    self.rebuild_tree(); // <-- this is new
} else {
    TreeUpdate {
        nodes: nodes.collect(),
        tree: None,
        focus: self.nodes.focus_node(&self.window_adapter_weak),
    }
}
```
